lessons/durablefunctions/chaining-ts.md

```diff
@@ -0,0 +1,631 @@
+# Azure Durable Functions - Introduction & Chaining
+
+## Goal 🎯
+
+The goal of this lesson is to give you an introduction into Azure Durable Functions including a first durable function that chains two functions calls. In addition we will take a look into some features of Durable Functions that help you when working with durable functions.
+
+This lessons consists of the following exercises:
+
+|Nr|Exercise
+|-|-
+|0| [Introduction to Azure Durable Functions](#0-introduction-to-azure-durable-functions)
+|0.1| [Functions and Chaining](#01-functions-and-chaining)
+|0.2| [Solution via Durable Functions](#02-solution-via-durable-functions)
+|0.3| [Mechanics of Durable Functions](#03-mechanics-of-durable-functions)
+|0.4| [Prerequisites for local development](#04-prerequisites-for-local-development)
+|1| [Creating a Function App project for a Durable Function](#1-creating-a-function-app-project-for-a-durable-function)
+|1.1| [The Client Function](#11-the-client-function)
+|1.2| [The Orchestrator Function](#12-the-orchestrator-function)
+|1.3| [The Activity Function](#13-the-activity-function)
+|1.4| [The First Execution](#14-the-first-execution)
+|2| [Implementing a "Real-World" Scenario](#2-implementing-a-"real-world"-scenario)

```

Comment:
So 2.x becomes 3.x. Although I do appreciate real world scenario's I feel introducing SAP is a bit too much. Most people will probably not be aware of what this is so this might be too much cognitive load. What about using the using Blob and Table bindings in the activities and perhaps use a 3rd party API that doesn't require a registration and is centered around a basic concept that is well understood by everyone (such as the homework example).

Reply:
I was a bit afraid of that ... but kind of expected. Here is a proposal for an alternative scenario making use of the GitHub API (no authentication needed, no API key needed at least up to a certain limit):
- Story: We want to search for a repo and get some information about the author
- Step1 (Activity 1): Call the repo search API of GitHub and extract the author from the result https://api.github.com//search/repositories/{query}
- Step 2 (Activity 2):  Call the users API to extract further data about the author 
https://api.github.com/users/{userID}

Would that new story be okay for you? If yes I will make the corresponding changes.

In addition  I would like to put the advanced scenario in an kind of "outlook section" or appendix as I want to drag at least some SAP folks attention to Azure functions :-)

Reply:
Yes the GitHub use case sounds good. 

What about an additional homework section with the existing SAP part in it? We can explain in the homework section this is a sort of stretch goal for the homework and we can still mention it in the recording.

Reply:
Additional homework section sounds fine to me. Then _this is the way_

Reply:
Done